### ISSUE TITLE
OfficeRuntime.auth.getAccessToken not working on production

### ISSUE THREAD

**Issue by arozwalak** [OfficeDev/office-js-docs-pr#2344]
I'm working on outlook addon using SSO. I've generated addon with yeoman. Locally everything works good, but once deployed to production and served built files from /dist SSO is not working anymore.

I found that locally this addon is running express.js server which is getting access token.
How I should handle this on production?

**Comment by TCourtneyOwen** [comment]
Perhaps this issue is similar to https://github.com/OfficeDev/office-js-docs-pr/issues/2279.  @davidchesnut is going to post documentation today that may help with this issue.

-Courtney

**Comment by davidchesnut** [comment]
Hi @arozwalak,

This issue is also related to https://github.com/OfficeDev/office-js-docs-pr/issues/2279. 

Since the code Courtney refers to is in a package, you can't update it to handle the Azure scenario. However you can get the same code from the [Office-Add-in-NodeJS-SSO](https://github.com/OfficeDev/Office-Add-in-NodeJS-SSO) sample. And you find the same code at https://github.com/OfficeDev/Office-Add-in-NodeJS-SSO/blob/master/SSOAutoSetup/src/configure/ssoAppDataSetttings.js (which you can modify.)

So it might make more sense to start with this code sample, and modify it to handle the secret for Azure. I'm still working with the engineering team to figure out the specific steps to modify this code for Azure deployment. Once I have more info I'll follow up here, and also update the docs.

Thanks,
David

**Comment by tobudim** [comment]
Hi @arozwalak , did you manage to deploy your add-in?

**Comment by arozwalak** [comment]
Hi @tobudim,
Unfortunately I can't help here. My addon is 'partially' developed, meaning that I just need accessToken, to authorize it with my platform. Looks like addon generated with office yeoman generator is not yet production ready. I was able to get around it and I'm able to authenticate user through our platform rather than through addon itself.

**Comment by tobudim** [comment]
@arozwalak thanks for your feedback, actually it very well guided me. 🙏

**Comment by arozwalak** [comment]
Glad I could help @tobudim  :)

**Comment by tobudim** [comment]
Hi @davidchesnut,

Update since my last message:

- Azure web app:
  - As we've seen getAccessToken doesn't work as it does not get the secret when built (`/dist`).
- Azure Windows Server VM:
  - As `npm start` works fine (I mirrored and updated the code to get the secret without asking for it locally) and not the build itself (`/dist`), I believed a reverse proxy would do fine.
  - As only Windows and Mac are supported I choose a Windows Server VM.
  - I could not go further than a 502 error, maybe or maybe not related to this repo but I could run other Node apps.

Bottom line: to me, I just can't use getAccessToken and I spend a lot of time building my app around this functionnality.

Any ideas would be appreciated, my goal is only to authenticate the user without asking him anything.

**Comment by davidchesnut** [comment]
Hi @dimitri-bourreau,

Just a quick update to let you know that I did get the [NodeJS SSO](https://github.com/OfficeDev/Office-Add-in-NodeJS-SSO) sample to deploy on Azure. In case this helps, here are [my notes]( https://github.com/OfficeDev/Office-Add-in-NodeJS-SSO/issues/61#issuecomment-821517972)

Cheers,
David

**Comment by davidchesnut** [comment]
Hopefully closing the loop on this now. We recently published documentation on how to deploy an SSO add-in from yo office. There were also updates made to yo office to better support deployment (be sure to use the latest office generator version.) For more details, see [Deploy a single sign-on (SSO) Office Add-in to Microsoft Azure App Service](https://learn.microsoft.com/office/dev/add-ins/publish/deploy-office-add-in-sso-to-azure)
